src/lib/common/topology_sampler.cpp

```diff
@@ -64,53 +88,102 @@ static void topology_sample_cb(networkTopology_t* networkTopology) {
 
     // compile all additional info here
     _node_list.num_nodes = networkTopology->length;
-    configASSERT(sizeof(_node_list.nodes) >= _node_list.num_nodes * sizeof(uint64_t)); // if we hit this, expand the node list size
-
-    network_crc32_calc = crc32_ieee_update(network_crc32_calc, reinterpret_cast<uint8_t *>(&_node_list.num_nodes), sizeof(_node_list.num_nodes));
+    // if we hit this, expand the node list size
+    configASSERT(sizeof(_node_list.nodes) >= _node_list.num_nodes * sizeof(uint64_t));
 
     memset(_node_list.nodes, 0, sizeof(_node_list.nodes));
 
     neighborTableEntry_t *cursor = NULL;
     uint16_t counter;
-    for(cursor = networkTopology->front, counter = 0; (cursor != NULL) && (counter < _node_list.num_nodes); cursor = cursor->nextNode, counter++) {
+    xQueueReset(_sys_info_queue);
+    xQueueReset(_config_cbor_map_queue);
+    bool exit = false;
+
+    // Iterate through the entire list of nodes and request the sys info.
+    for (cursor = networkTopology->front, counter = 0;
+         (cursor != NULL) && (counter < _node_list.num_nodes);
+         cursor = cursor->nextNode, counter++) {
       _node_list.nodes[counter] = cursor->neighbor_table_reply->node_id;
-      network_crc32_calc = crc32_ieee_update(network_crc32_calc, reinterpret_cast<uint8_t *>(&_node_list.nodes[counter]), sizeof(uint64_t));
+      if (!sys_info_service_request(_node_list.nodes[counter], sys_info_reply_cb,
+                                    NODE_SYS_INFO_REQUEST_TIMEOUT_S)) {
+        printf("Failed to send sys info request to node: %" PRIu64 "\n",
+               _node_list.nodes[counter]);
+        exit = true;
+        break;
+      }
+    }
+    if (exit) {
+      break;
     }
 
-    bm_common_config_crc_t config_crc = {
-      .partition = BM_COMMON_CFG_PARTITION_SYSTEM,
-      .crc32 = _sys_cfg->getCborEncodedConfigurationCrc32(),
-    };
-
-    network_crc32_calc = crc32_ieee_update(network_crc32_calc, reinterpret_cast<uint8_t *>(&config_crc), sizeof(bm_common_config_crc_t));
-
-    bm_common_fw_version_t fw_info = {
-      .major = 0,
-      .minor = 0,
-      .revision = 0,
-      .gitSHA = 0,
-    };
-
-    getFWVersion(&fw_info.major, &fw_info.minor, &fw_info.revision);
-    fw_info.gitSHA = getGitSHA();
+    // Create the network info cbor array and calculate the crc32
+    size_t cbor_bufsize =
+        _node_list.num_nodes * (sizeof(SysInfoSvcReplyMsg::Data) +
+                                sizeof(ConfigCborMapSrvReplyMsg::Data) + NODE_CONFIG_PADDING);
```

Comment:
Right now, for our current use cases, `NODE_CONFIG_PADDING` seems too big. However, future use cases will definitely require more space as Bristlemouth networks have more nodes and more system config values set. The example in the PR description only needed 142 bytes but this would have allocated 1680. (I believe each of the msg data types would be 24 bytes.)

My main question is how much heap do we have available? If we left this as is, allocating 560 for each node, how many nodes would use up our heap? Any idea? No big deal for DVT2 since 3 is the max we expect in this phase. Just wanted to share my 🤔 .

Reply:
On the order of 126k
```Available:            126216```

So plenty for now, I can reduce the padding to 128 if that's a more reasonable ceiling.

Reply:
Oh wow, that's more than I expected. Just leave it. 👍